ACKReactiveExtensions/Marshal/MarshalMapping.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  MarshalMapping.swift
+//  ACKReactiveExtensions
+//
+//  Created by Jakub Olejník on 06/04/2017.
+//  Ackee
+//
+
+import Result
+import Marshal
+import ReactiveSwift
+
+/**
+ * Protocol that allows creation of custom Marshal errors
+ */
+public protocol MarshalErrorCreatable: Error {
+    
+    /**
+     * Create error containing passed `MarshalError`
+     *
+     * - parameter marshalError: `MarshalError` which should be wrapped
+     */
+    static func createMarshalError(_ marshalError: MarshalError) -> Self
+}
+
+extension MarshalError: MarshalErrorCreatable {
+    public static func createMarshalError(_ marshalError: MarshalError) -> MarshalError {
```

Comment:
Now when all errors thrown by the extension are really `MarshalError`s this is correct.